packages/client/webapp/src/components/Sidebar.vue

```diff
@@ -0,0 +1,134 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+import Button from '@/components/Button.vue';
+import SimulationPlan from '@carbon/icons-vue/es/data-player/32';
+import MachineLearningModel from '@carbon/icons-vue/es/machine-learning-model/32';
+import Dataset from '@carbon/icons-vue/es/table--split/32';
+import ProvenanceGraph from '@carbon/icons-vue/es/flow/32';
+import Search from '@carbon/icons-vue/es/search--locate/32';
+import User from '@carbon/icons-vue/es/user/32';
+
+const props = defineProps<{ sidebarPosition: String }>();
+const emit = defineEmits(['updateSidebarPosition']);
+
+const mode = ref('Simulation Plan');
```

Comment:
I would recommend extracting 'Simulation Plan' and the other hardcoded strings to a constant enum at the top of the file. This enables auto-complete and static checking and avoids typos when repeating these strings in multiple places.

Reply:
Great point Ben, definitely want to avoid repeating yourself. Stay DRY everyone ;)

One followup question. Does/Will the app need language support? If so might be good to get started on localization of all these strings as well.

Reply:
https://www.typescriptlang.org/docs/handbook/enums.html

Reply:
@Nithos My guess would be not to worry about localization any time soon. In world modelers it never once became a requirement, and setting up a language support system in the meantime would add overhead when writing and understanding code.